### ISSUE TITLE
Some users have deleted their location

### ISSUE THREAD

**Issue by UserGoodtm** [grokability/snipe-it#13251]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

The other day I updated to the latest version 6.1.1 from 6.1.0. And today I found that some users have deleted the location. I found out experimentally that this was due to synchronization with LDAP (AD). My setup doesn't use the location field for synchronization, so it shouldn't have affected in any way, but it did.

### Reproduction steps

1. Updated to the latest version 6.1.1
2. I didn't make any changes in the LDAP settings
3. Synchronized users with AD (LDAP)
4. Many users have deleted the location
...

### Expected behavior

Changes should only occur with direct instructions to do so. If there are empty fields in the parameter settings, without instructions, then changes to these parameters should not occur

### Snipe-IT Version

v6.1.1 build 10847

### Operating System

CentOS

### Web Server

Apache

### PHP Version

7.4.29

### Operating System

Windows

### Browser

All

### Version

Last

### Additional context

Is this a fresh install or an upgrade? - upgrade

**Comment by snipe** [comment]
Do you have any value in the LDAP Location field in your LDAP settings?

<img width="883" alt="Screenshot 2023-07-05 at 4 07 10 PM" src="https://github.com/snipe/snipe-it/assets/197404/5eecedbf-2d9f-46e1-aa0e-2e04b4b0929a">

**Comment by xinput** [comment]
We have the same issue on our instance after updating to 6.1.1
Also an update to 6.1.2 did not fixed the issue.

Another  UAT instance running on 6.1.0 does not have this issue.

it seems not all users are affected, it's like around 60-70% of users where location has been deleted. Also checked the database, their Location ID is set to NULL.

@snipe  LDAP location field in LDAP Setting is empty also checked in settings table in DB and verified the value of ldap_location is NULL

**Comment by UserGoodtm** [comment]
> Do you have any value in the LDAP Location field in your LDAP settings?
> 
> <img alt="Screenshot 2023-07-05 at 4 07 10 PM" width="883" src="https://user-images.githubusercontent.com/197404/251194561-5eecedbf-2d9f-46e1-aa0e-2e04b4b0929a.png">

No. There is no data in the input field. The field is empty. By meaning, if the field is empty, then no actions (synchronization) should be performed

**Comment by maur27-it** [comment]
We have the same issue. I updated to v6.1.1 and after LDAP sync all locations for active users are deleted. So in our case this issue affects all active users. I also tried setting the same location name in AD and in Snipe-IT, but it doesn't work. In previous version v6.0.14 LDAP sync didn't delete the locations and everything works correctly. 

The "LDAP Location" field is empty. And I didn't find any other property for LDAP location sync.
